### PR TITLE
Upgrade revealjs to 4.0.2

### PR DESCRIPTION
**Pull request by veggiemonk**
Hi, first of all, thank you for this awesome project!!
I converted all my presentations to reveal-hugo.

I saw #90 and I need your help to improve this PR. 
I checked and everything seems to work properly but I have probably missed something.

steps taken:
* used the `/dist/` folder of the revealjs repo
* removed the print-pdf css reference (automatically included in `reveal.css`)
* moved plugins loading before initialization of `Reveal`

Let me know everything that can be improved or that you would like to see :-)

### CODE REVIEW

**Comment by dzello** [comment]
@veggiemonk Thanks for putting the PR together! I'll set aside a few minutes later this week to review. Cheers.

**Comment by Chan9390** [comment]
Good PR

**Comment by ccamara** [comment]
Hello there, @dzello hope you are doing fine. Did you happen to find some time to merge this PR? I'd love to see this feature implemented.

**Comment by dzello** [comment]
@ccamara Thanks for the ping. I've struggled to find time to review it but I'd love to merge this in. I know you're very familiar with the project and have contributed a lot before - by chance do you have a minute to test this PR and confirm that everything is working?

@veggiemonk Thanks for being patient, we'll get this handled soon!

**Comment by ccamara** [comment]
@dzello I'll try to do so, sure! (probably on weekend, though).

**Comment by ccamara** [comment]
TLDR: Dear all, I have reviewed this PR and while most of the things work well, I have found some problems with custom themes and code line numbering.

Full version:

Since it is my first PR revision in this repo (or for hugo) and I am not an expert at all neither with hugo nor with reveal (nor javascript in general), let me explain step by step what I have done:

1. Created a new hugo site from scratch
2. Added reveal-hugo theme as submodule as explained in the readme file
3. Created a branch with the changes made in this PR (it has to be noted that since it was made some time ago, it is now one commit below the master branch here).
4. Copied `exampleSite`'s content to hugo site's root, so I could get some default content.
5. Generated the site with `hugo-serve`
6. Reviewed all slides within `exampleSite/content` and compared with the results without appliying this PR
7. Copied my own slides from my repo and compared with live site

That's what I found so far:

1. Slides with custom theming fail to apply all the rules (such as background). This is the case of `custom-theme-example` and my own slides (all of them use a custom scss file).
2. Code line numbers are not displayed on `content/highlightjs-linenumbers-example/`. Funny enough, line numbers are shown this other slide deck: `/hugo-hl-example/#/4` (

To be tested:

1. New reveal 4.x features, such as auto-animate (the one I am most interested with so far).

What I cannot assess:

I see this PR makes quite some changes in `static/reveal-js` file names, structure and content, which I do not have enough knowledge to assess. At first I thought that would be following the [upgrade from 3.x to 4.x instructions on official documentation](https://revealjs.com/upgrading/), but I was wrong, as I haven't been able to find those specific changes and there are even more. Surely @veggiemonk has an explanation for that

**Comment by veggiemonk** [comment]
Hi @ccamara 

Thank you for testing.

I did not follow the upgrade guide as far as I can remember.

steps taken:
* used the `/dist/` folder of the revealjs repo
* removed the print-pdf css reference (automatically included in `reveal.css`)
* moved plugins loading before initialization of `Reveal`

It was a straightforward delete of the previous version of revealjs and a dump of the production file of the new revealjs.
Modified some plugins initialization. 
Check if my site was working properly.
Find the time to make a PR so that other may benefit from it.

I'll look at it this weekend.

**Comment by ccamara** [comment]
Great, @veggiemonk , thank you again for sharing your solution and your time!

**Comment by veggiemonk** [comment]
Hi @ccamara @dzello 

All should now be fixed. 

Regarding 1. The asset were missing. They are now added.

Regarding 2.
For the line number, they are explicitly hidden, see `exampleSite/layouts/partials/highlightjs-linenumbers-example/reveal-hugo/body.html`

Let me know what you think,

**Comment by ccamara** [comment]
Great, @veggiemonk ! Thank you very much!
Unfortunately, I am not sure if it is because it's late and I'm sleepy, but after updating repo with your last commit, slides are not working at all.
I can't see any content, nor the navigation arrows. In fact, I can't navigate through the slides. I'll check more thoroughly on weekend or end of next week.

**Comment by guilhem** [comment]
Any news? :wave:

**Comment by madawei2699** [comment]
Any update?

**Comment by veggiemonk** [comment]
@dzello Is there something you need from me to merge this ?

**Comment by chrisreddington** [comment]
@veggiemonk, @dzello - I'm interested to know the latest on this PR. I've been looking around at different options for Markdown Presentation generators. As i'm using Hugo already, I thought that  this theme could be a great option :)

It looks like reveal-hugo is using reveal.js v3.9.2, whereas the latest reveal.js is v4.2.1.

Any status on updates/plans would be great! thank you!

**Comment by chrisreddington** [comment]
FYI - I have made a separate pull request (#115) to take the dependency to 4.2.1

**Comment by davidovich** [comment]
@dzello This can be closed now that #146 is in. Thanks!
